pkg/sql/opt/norm/scalar_funcs.go

```diff
@@ -384,3 +386,14 @@ func (c *CustomFuncs) SplitTupleEq(lhs, rhs *memo.TupleExpr) memo.FiltersExpr {
 func (c *CustomFuncs) CanNormalizeArrayFlatten(input memo.RelExpr, p *memo.SubqueryPrivate) bool {
 	return c.HasOuterCols(input) || p.WithinUDF
 }
+
+func (c *CustomFuncs) CollapseDupeChar(input opt.ScalarExpr) opt.ScalarExpr {
+	pattern := memo.ExtractConstDatum(input).String()
```

Comment:
We don't want to use the `String()` method here, since this formats the `DString` which is costly and unnecessary. Instead, we should use a type assertion:

```go
	s, ok := memo.ExtractConstDatum(cnst).(*tree.DString)
	if !ok {
		panic(errors.AssertionFailedf("expected string constant")
	}
```